indico/queries/submission.py

```diff
@@ -0,0 +1,194 @@
+                "workflowIds": workflow_ids,
+                "filters": filters,
+                "limit": limit,
+            },
+        )
+
+    def process_response(self, response) -> List[Submission]:
+        return [
+            Submission(**s)
+            for s in super().process_response(response)["submissions"]["submissions"]
+        ]
+
+
+class GetSubmission(GraphQLRequest):
+    """
+    Retrieve a Submission by id
+
+    Args:
+        submission_id (int): Submission id
+
+    Returns:
+        Submission: Found Submission object
+    """
+
+    query = """
+        query GetSubmission($submissionId: Int!){
+            submission(id: $submissionId){
+                id
+                datasetId
+                workflowId
+                status
+                inputFile
+                inputFilename
+                resultFile
+            }
+        }
+    """
+
+    def __init__(
+        self, submission_id: int,
+    ):
+        super().__init__(
+            self.query, variables={"submissionId": submission_id},
+        )
+
+    def process_response(self, response) -> Submission:
+        return Submission(**(super().process_response(response)["submission"]))
+
+
+class GenerateSubmissionResult(GraphQLRequest):
+    query = """
+        mutation CreateSubmissionResults($submissionId: Int!) {
+            submissionResults(submissionId: $submissionId) {
+                jobId
+            }
+        }
+
+    """
+
+    def __init__(
+        self, submission_id: int
+    ):
+        super().__init__(
+            self.query, variables={"submissionId": submission_id},
+        )
+
+    def process_response(self, response) -> Job:
+        response = super().process_response(response)["submissionResults"]
+        return Job(id=response["jobId"])
+
+
+class SubmissionResult(RequestChain):
+    """
+    Generate a result file for a Submission
+
+    Args:
+        submission_id (int): Id of the submission
+        check_status (str, optional): Submission status to check for.
+            Defaults to any status other than `PROCESSING`
+        wait (bool, optional): Wait until the submission is `check_status`
+            and wait for the result file to be generated. Defaults to False
+        timeout (int or float, optional): Maximum number of seconds to wait before
+            timing out. Ignored if not `wait`. Defaults to 30
+
+    Returns:
+        If `wait`:
+            str: URL to result file that can be retrieved with `RetrieveStorageObject`
+
+        If not `wait`:
+            Job: Job that can be watched for results
+
+    Raises:
+        if `wait`:
+            IndicoTimeoutError: Submission was not `check_status`
+               `or result file Job did not complete within `timeout` seconds.
+        If not `wait`:
+            IndicoInputError: The requested Submission is not `check_status`
+    """
+
+    previous: Submission = None
+
+    def __init__(
+        self,
+        submission_id: int,
```

Comment:
same as above the interface could be a little nicer if we just took the submission object